### PR TITLE
iss1466 RandomNumber Correction for bounds of 0

### PR DESCRIPTION
**Pull request by loganom**
If the start or end value was 0, then no matter what, the default rand() value was returned (which is a fractional number). I added a check for the difference of the two numbers. If it's greater than 1, then we enter the random number if block.

Added a few tests to show this.

Here's to hoping I've done PR creation correctly.. it's been a while.

#1466

### CODE REVIEW

**Comment by loganom** [comment]
@null-minus-null  @moollaza

**Comment by loganom** [comment]
Should I also add myself to the attribution list?

**Comment by mintsoft** [comment]
@loganom no reason not to add yourself :)

File: lib/DDG/Goodie/RandomNumber.pm [hunk 1]
```diff
@@ -37,9 +37,11 @@ handle query_lc => sub {
 
     ($end, $start) = ($start, $end) if ($start > $end);
 
+    my $valDiff = $end - $start;
+
     my $rand = rand;
 
-    if ($start && $end) {
+    if ($start && $end || $valDiff > 1) {
         $rand *= ($end - $start + 1);
```
Comment (mintsoft) [inline lib/DDG/Goodie/RandomNumber.pm]: I'd probably reuse `$valDiff` here, so it'd be `$rand *= $valDiff +1 ;`

**Comment by mintsoft** [comment]
@loganom Looks good to me, if you want to make that tiny tweak and add yourself to the attribution I'll marge when done :+1:

**Comment by loganom** [comment]
@mintsoft thanks for the comments. Good to go.

**Comment by mintsoft** [comment]
Thanks for the fix!

**Pull request merged**